pkg/acceptance/testenvs/testenvs_test.go

```diff
@@ -0,0 +1,82 @@
+package testenvs_test
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/testenvs"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_GetOrSkipTest(t *testing.T) {
+	runGetOrSkipInGoroutineAndWait := func(t *testing.T) string {
```

Comment:
you also have this defined below. should this be refactored out into a helper function

Reply:
There are differences between them (e.g. returning value). I don't think that this will help with readability nad that this is necessary.